src/vite/observable.ts

```diff
@@ -45,7 +62,10 @@ export function observable({
       order: "pre",
       async handler(input, context) {
         const notebook = deserialize(input, {parser});
-        const tsource = await readFile(template, "utf-8");
+        let tsource = await readFile(template, "utf-8");
+        if (transformTemplate) {
+          tsource = await transformTemplate(tsource, context);
+        }
```

Comment:
I prefer to be more exacting in public APIs (no reason to allow `transformTemplate: false`):

```suggestion
        if (transformTemplate !== undefined) {
          tsource = await transformTemplate(tsource, context);
        }
```

Allowing null would be fine, too, but then I’d make the `ObservableOptions["transformTemplate"]` also allow null.